pkg/db/file_impl.go

```diff
@@ -56,4 +60,124 @@ func (F *FSBackend) Files() ([]string, error) {
 	return names, nil
 }
 
+func jsonFromFile(filename string, i interface{}) (err error) {
+	jsonFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(jsonFile, &i)
+
+	return err
+}
+func jsonToFile(filename string, i interface{}) error {
+	newJson, err := json.Marshal(i)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filename, newJson, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (F *FSBackend) getCategoriesFromFile() (Categories, error) {
+	var cat Categories
+	err := jsonFromFile(path.Join(F.dirLocation, "cats.json"), &cat)
+	return cat, err
+}
+
+func (F *FSBackend) writeCategoriesToFile(cats Categories) error {
+	err := jsonToFile(path.Join(F.dirLocation, "cats.json"), cats)
+	return err
+}
+
+func (F *FSBackend) CreateCategory(name string) (CategoryID, error) {
+	// todo: category names should be unique (across documents?) either way,
+	//  to be enforced by the db via constraints (check those errors)
+	F.catMutex.Lock()
+	defer F.catMutex.Unlock()
+	cats, err := F.getCategoriesFromFile()
+	if err != nil {
+		if os.IsNotExist(err) { // no file? no problem, we're about to write to it
+			cats = Categories{Categories: map[string]Category{}}
+		} else {
+			return "", err
+		}
+	}
+
+	// since the ID is just going to be the name (until there's a db providing AI,
+	//  use the name as the ID. Therefor, no point in check if the name already exists
+	newCat := Category{Name: name, ID: name, Texts: []DocumentText{}}
+
+	cats.Categories[name] = newCat
+
+	err = F.writeCategoriesToFile(cats)
+	if err != nil {
+		return "", err
+	}
+
+	return newCat.ID, nil
+}
+
+func (F *FSBackend) CategorizeText(categoryID CategoryID, documentID FileID, text string) error {
+	F.catMutex.Lock()
+	defer F.catMutex.Unlock()
+	cats, err := F.getCategoriesFromFile()
+	if err != nil {
+		return err
+	}
+
+	if _, ok := cats.Categories[categoryID]; ok == false {
+		return errors.New("No category found with ID: " + categoryID)
+	}
+
+	var cat = cats.Categories[categoryID]
+	cat.Texts = append(cat.Texts, DocumentText{
+		DocumentID: documentID,
+		Text:       text,
+	})
+	cats.Categories[categoryID] = cat
+
+	err = F.writeCategoriesToFile(cats)
+	return err
+}
+
+func (F *FSBackend) GetCategory(categoryID CategoryID) (Category, error) {
+	F.catMutex.RLock()
+	defer F.catMutex.RUnlock()
+	cats, err := F.getCategoriesFromFile()
+	if err != nil {
+		return Category{}, err
+	}
+
+	if cat, ok := cats.Categories[categoryID]; ok {
+		return cat, nil
+	}
+
+	// todo: is this how "no category found" should be returned?
+	return Category{}, nil
```

Comment:
Think its probably an error? Since that means that some how the consumer has an ID, but its invalid. They need to be notified of this.